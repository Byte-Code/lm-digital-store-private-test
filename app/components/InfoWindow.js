import React from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import BlockIcon from 'material-ui/svg-icons/navigation/close';
import glamorous from 'glamorous';

import { getStockLabel } from '../utils/utils';

const Wrapper = glamorous.div({
  width: '300px',
  backgroundColor: '#fff',
  boxShadow: '0 0 8px 0 #e4e4e4',
  position: 'absolute',
  top: '-230px',
  left: 0,
  padding: '20px',
  zIndex: 1
});

const Divider = glamorous.div({
  border: '1px dashed #333333',
  width: '100%'
});

const Stock = glamorous.div(({ isAvailable }) => ({
  marginTop: '15px',
  color: isAvailable ? '#67cb33' : '#333333',
  fontSize: '14px',
  fontFamily: 'LeroyMerlinSans Bold'
}));

const StoreName = glamorous.p({
  fontSize: '20px',
  color: '#262626',
  marginBottom: '10px',
  fontFamily: 'LeroyMerlinSans Bold'
});

const ArrowDown = glamorous.div({
  width: 0,
  height: 0,
  borderLeft: '14px solid transparent',
  borderRight: '14px solid transparent',
  borderTop: '14px solid #fff',
  position: 'absolute',
  bottom: '-14px',
  left: '14px',
  filter: 'drop-shadow(0 2px 0 #e4e4e4)'
});

const Address = glamorous.div({
  marginBottom: '20px',
  '&>p': {
    fontSize: '15px',
    color: '#333333'
  }
});

const iconStyle = {
  width: 24,
  height: 24,
  position: 'absolute',
  top: 5,
  right: 5,
  cursor: 'pointer'
};

const InfoWindow = ({ currentStoreInfo, handleClick }) => {
  const name = currentStoreInfo.get('name');
  const availability = currentStoreInfo.get('storeStock');
  const isAvailable = availability > 0;
  const label = getStockLabel(availability);
  const street = currentStoreInfo.getIn(['address', 'street']);
  const streetNumber = currentStoreInfo.getIn(['address', 'streetNumber']) || '';
  const zip = currentStoreInfo.getIn(['address', 'zipCode']);
  const city = currentStoreInfo.getIn(['address', 'city']);
  const province = currentStoreInfo.getIn(['address', 'state']);

  return (
    <Wrapper>
      <BlockIcon style={iconStyle} fill="#333333" onTouchTap={handleClick} />
      <div>
        <StoreName>{name}</StoreName>
        <Address>
          <p>{`${street} ${streetNumber}`}</p>
          <p>{`${city} - (${province})`}</p>
          <p>{zip}</p>
        </Address>
      </div>
      <Divider />
      <Stock isAvailable={isAvailable}>
        {label}
      </Stock>
      <ArrowDown />
    </Wrapper>
  );
};

InfoWindow.propTypes = {
  currentStoreInfo: ImmutablePropTypes.map.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default InfoWindow;
