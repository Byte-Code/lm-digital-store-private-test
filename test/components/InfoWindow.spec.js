import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';
import BlockIcon from 'material-ui/svg-icons/navigation/close';

import InfoWindow from '../../app/components/InfoWindow';

const currentStoreInfo = fromJS({
  code: '123',
  name: 'abc',
  latitude: 1231,
  longitude: 123213,
  address: {
    state: 'abc',
    street: 'bde',
    streetNumber: '1',
    zipCode: '213',
    city: 'milan'
  }
});
const handleClick = jest.fn();

describe('InfoWindow', () => {
  const wrapper = shallow(
    <InfoWindow
      handleClick={handleClick}
      currentStoreInfo={currentStoreInfo}
    />
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleClick when BlockIcon is clicked', () => {
    wrapper.find(BlockIcon).simulate('touchTap');
    expect(handleClick).toHaveBeenCalled();
  });
});
