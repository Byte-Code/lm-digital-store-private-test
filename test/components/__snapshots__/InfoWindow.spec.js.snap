// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoWindow should render properly 1`] = `
<glamorous(div)>
  <NavigationClose
    fill="#333333"
    onTouchTap={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "height": 24,
        "position": "absolute",
        "right": 5,
        "top": 5,
        "width": 24,
      }
    }
  />
  <div>
    <glamorous(p)>
      abc
    </glamorous(p)>
    <glamorous(div)>
      <p>
        bde 1
      </p>
      <p>
        milan - (abc)
      </p>
      <p>
        213
      </p>
    </glamorous(div)>
  </div>
  <glamorous(div) />
  <glamorous(div)
    isAvailable={false}
  >
    Prodotto disponibile su ordinazione
  </glamorous(div)>
  <glamorous(div) />
</glamorous(div)>
`;
